api/bindings/v1alpha1/endpointbinding_types.go

```diff
@@ -33,6 +33,15 @@ import (
 
 // EndpointBindingSpec defines the desired state of EndpointBinding
 type EndpointBindingSpec struct {
+	// FQRI "Fully Qualified Resource Identifier" is the unique identifier
+	// representing the EndpointBinding + its Endpoints
+	// Format: <scheme>://<service>.<namespace>:<port>
+	//
+	// +kubebuilder:validation:Required
+	// See: https://regex101.com/r/9QkXWl/1
+	// +kubebuilder:validation:Pattern=`^((?P<scheme>(tcp|http|https|tls)?)://)?(?P<service>[a-z][a-zA-Z0-9-]{0,62})\.(?P<namespace>[a-z][a-zA-Z0-9-]{0,62})(:(?P<port>\d+))?$`
+	FQRI string `json:"fqri"`
```

Comment:
Nit: I struggled to come up with a name for this field, but I really do want to store this data because the whole parsing hostport thing should only really happen once.

Thoughts on a name? FQRI is what I came up with in the shower haha

Reply:
This feels very much like [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.2). Off the cuff, I would go with URI or something similar. I'll think on it though as well.